### PR TITLE
scripts: Improve handling of API variants

### PR DESCRIPTION
**Pull request by aqnuep**
This PR introduces some improvements to the generator scripts to better handle API variants defined in the vk.xml, as the previous code did not handle some of the aspects of the new vk.xml syntaxes introduced for API variant dependent definitions in a general way.

### CODE REVIEW

**Comment by christophe-lunarg** [comment]
@aqnuep, I think I would feel a little more confident about this if I had a better idea of what specific cases it resolves.

For example, could you had some tests, maybe adding "profiles" here E:\Github\khronos\Vulkan-Profiles - Bugfix\profiles\test\data and manually checking the specific cases in E:\Github\khronos\Vulkan-Profiles - Bugfix\layer\tests\tests_mechanism.cpp

?

File: scripts/gen_profiles_solution.py [hunk 1]
```diff
@@ -1421,7 +1455,7 @@ def __init__(self, xml, upperCaseName):
         self.limits = dict()
         self.platform = xml.get('platform')
         self.provisional = xml.get('provisional')
-        self.promotedTo = xml.get('promotedto')
+        self.promotedTo = xml.get('promotedto').split(',') if xml.get('promotedto') is not None else []
```
Comment (christophe-lunarg) [inline scripts/gen_profiles_solution.py:1458]: Is there a case where an extension is promoted to multiple extensions? Or is this for promotedto an extension and a core version maybe?

**Review by christophe-lunarg — APPROVED** [review]

**Comment by christophe-lunarg** [comment]
I think the changes are mostly to better handle Vulkan and Vulkan SC it seems reasonnable.

It looks like it also fix some cases regarding promotedto and alias for which I could enjoy seeing additional tests to prevent future changes accidently break these cases.

**Comment by aqnuep** [comment]
Yes, most of these were detected by testing against Vulkan SC content, but are rather generic fixes so far.

The problem is that in order to write tests for this you'd actually have to create custom vk.xml inputs to test with, e.g. one where `supported="vulkan,vulkansc"` is replaced with `supported="vulkansc,vulkan"` in the extension tags, which the code before this PR doesn't handle.

**Pull request merged**